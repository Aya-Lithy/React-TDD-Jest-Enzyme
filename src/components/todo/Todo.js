import React from "react";

const Todo = ({ todo }) => {
  const { id, title, completed } = todo;
  const h4 = <h4>{title}</h4>;
  const text = completed ? <strike>{h4}</strike> : h4;

  return <div data-testid={`todo-${id}`}>{text}</div>;
};

export default Todo;
