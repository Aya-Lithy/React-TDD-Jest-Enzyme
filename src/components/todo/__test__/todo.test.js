import { render, screen } from "@testing-library/react";
import Todo from "../todo";
import renderer from "react-test-renderer";

test("should render non-completed todo", () => {
  const todo = { id: 1, title: "prepare breakfast", completed: false };
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("prepare breakfast");
  expect(todoElement).not.toContainHTML("<strike>");
});

//snapshot testing
test("matches snapshot", () => {
  const todo = { id: 1, title: "prepare breakfast", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  expect(tree).toMatchSnapshot();
});
