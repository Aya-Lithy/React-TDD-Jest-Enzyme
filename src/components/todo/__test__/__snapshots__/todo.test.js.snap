// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h4>
    prepare breakfast
  </h4>
</div>
`;
